pytx/pytx/common.py

```diff
@@ -0,0 +1,345 @@
+    def objects(cls, text=None, strict_text=False, type_=None,
+                limit=None, since=None, until=None, __raw__=None,
+                full_response=False, dict_generator=False):
+        """
+        Get objects from ThreatExchange.
+
+        :param text: The text used for limiting the search.
+        :type text: str
+        :param strict_text: Whether we should use strict searching.
+        :type strict_text: bool, str, int
+        :param type_: The Indicator type to limit to.
+        :type type_: str
+        :param limit: The maximum number of objects to return.
+        :type limit: int, str
+        :param since: The timestamp to limit the beginning of the search.
+        :type since: str
+        :param until: The timestamp to limit the end of the search.
+        :type until: str
+        :param __raw__: Provide a dictionary to force as GET parameters.
+                        Overrides all other arguments.
+        :type __raw__: dict
+        :param full_response: Return the full response instead of the generator.
+                              Takes precedence over dict_generator.
+        :type full_response: bool
+        :param dict_generator: Return a dictionary instead of an instantiated
+                               object.
+        :returns: Generator, dict (using json.loads())
+        """
+
+        if __raw__:
+            if isinstance(__raw__, dict):
+                params = __raw__
+            else:
+                raise pytxValueError("__raw__ must be of type dict")
+        else:
+            params = init.Broker.build_get_parameters(
+                text=text,
+                strict_text=strict_text,
+                type_=type_,
+                limit=limit,
+                since=since,
+                until=until,
+            )
+        if full_response:
+            return init.Broker.get(cls._URL, params=params)
+        else:
+            return init.Broker.get_generator(cls,
+                                             cls._URL,
+                                             limit,
+                                             to_dict=dict_generator,
+                                             params=params)
+
+    @class_or_instance_method
+    def details(cls_or_self, id=None, fields=None, connection=None,
+                full_response=False, dict_generator=False, metadata=False):
+        """
+        Get object details. Allows you to limit the fields returned in the
+        object's details. Also allows you to provide a connection. If a
+        connection is provided, the related objects will be returned instead
+        of the object itself.
+
+        NOTE: This method can be used on both instantiated and uninstantiated
+        classes like so:
+
+            foo = ThreatIndicator(id='1234')
+            foo.details()
+
+            foo = ThreatIndicator.details(id='1234')
+
+
+        BE AWARE: Due to the nature of ThreatExchange allowing you to query for
```

Comment:
Definitely a feature, but worth noting in detail for developers who are confused when they use an ID in the wrong class and wonder why things look messed up. We could get around this if objects returned their type by default without having to query with `&metadata=1` and parsing out the type from there.